Replicator/Replicator.cc

```diff
@@ -147,6 +147,95 @@ namespace litecore { namespace repl {
         _disconnect(websocket::kCodeNormal, {});
     }
 
+    alloc_slice Replicator::pendingDocumentIDs(C4Error* outErr) const {
+        if(!_pusher) {
+            // Couchbase Lite should not allow this case
+            return nullslice;
+        }
+
+        return _db->use<alloc_slice>([this, outErr](C4Database* db)
+        {
+            const auto dbLastSequence = c4db_getLastSequence(db);
+            const auto replLastSequence = _checkpoint.sequences().local;
+            if(replLastSequence >= dbLastSequence) {
+                return alloc_slice(nullslice);
```

Comment:
is this how an empty encoded fleece array created?

Reply:
No, this is a null value.  It's easier than encoding an empty array (which needs an FLEncoder)

Reply:
but then whats the difference between `return nullslice;` and `return alloc_slice(nullslice);`

Reply:
Just to satisfy the return type of this method since `nullslice` is not of type `alloc_slice`